scrapy/downloadermiddlewares/robotstxt.py

```diff
@@ -83,21 +88,11 @@ def _logerror(self, failure, request, spider):
         return failure
 
     def _parse_robots(self, response, netloc):
-        rp = robotparser.RobotFileParser(response.url)
         body = ''
-        if hasattr(response, 'text'):
-            body = response.text
-        else: # last effort try
-            try:
-                body = response.body.decode('utf-8')
-            except UnicodeDecodeError:
-                # If we found garbage, disregard it:,
-                # but keep the lookup cached (in self._parsers)
-                # Running rp.parse() will set rp state from
-                # 'disallow all' to 'allow any'.
-                pass
-        # stdlib's robotparser expects native 'str' ;
-        # with unicode input, non-ASCII encoded bytes decoding fails in Python2
+        policy = HeaderWithDefaultPolicy(default=1800, minimum=600)
+
+        rp=Robots.fetch(response.url, ttl_policy=policy)
```

Comment:
I understand that this performs an HTTP request to get the robots.txt file. Can you check?
We do not want to re-fetch the file, nor use a synchronous networking call (reppy uses python-requests as far as I understand)